### ISSUE TITLE
HEIC Support

### ISSUE THREAD

**Issue by dannyhanes** [BookStackApp/BookStack#2505]
**Describe the feature you'd like**
Add support for HEIC images

**Describe the benefits this feature would bring to BookStack users**
Not sure how many users run into this, but before I can upload photos to Bookstack, I would need to convert them to a compatible format. This would just allow for easier uploading of iPhone images.

**Comment by ssddanbrown** [comment]
Thanks for the request @dannyhanes, (And also thanks for the support via sponsor!),

Ideally, to be supported, we'd need our [image resizing library](http://image.intervention.io/getting_started/formats) to support these files otherwise it'll be tricky to be able to display these in the gallery or auto-create reduced sized versions for page content.

In addition, It seems that general support for the file format is [still pretty weak](https://caniuse.com/heif) with it potentially dependant on people having installed system extensions/packages to support the format. It's a [similar case with AVIF](https://caniuse.com/avif) although that may some sooner thanks to its openess.

I seem to remember something about iOS auto-converting files to jpeg upon upload to web? [Apple page lightly](https://support.apple.com/en-gb/HT207022) mentions something along these lines. Might have to power up the old iPhone SE and have a play.

**Comment by dannyhanes** [comment]
@ssddanbrown Yeah, when I put the request in, I kinda figured with it being new'ish and generally an Apple only feature, that it would probably be low on the list. I just noticed it when I went to upload some photos and thought at the very least I could make a request. 

And I am glad I am able to support Bookstack. :) Honestly, it is probably the most used self-hosted app in my house. Keep up the great work!

**Comment by ssddanbrown** [comment]
Just checking in on this, Kinda also tracking AVIF as part of this since it's a similar situation (And since they're somewhat related via HEIF).

### AVIF

- System: Looks to now be supported to some level, although potentially only on PHP 8.1 and greater.
- Browser: Support has improved, although Safari dragging everyone back.

### HEIC

- System:
  - [Not yet supported by GD](https://github.com/Intervention/image/blob/a1abc986fd644365e1740e70ea89b43b39c5cd59/src/Intervention/Image/Gd/Encoder.php#L174-L179)
  - [Could technically be supported by ImageMagik](https://github.com/Intervention/image/blob/a1abc986fd644365e1740e70ea89b43b39c5cd59/src/Intervention/Image/Imagick/Encoder.php#L218-L227) but requires some quite specific and complex imagemagik building to get imagemagik to support the format.
- Browser: Still pretty bad, Currently nothing appears to support them, even Safari doesn't seem to yet support the format from what I can find.

### Thoughts

AVIF is coming along quite nicely although PHP 8.1 support requirement will be a restriction. I'm thinking early/mid next year would be a good point to add support (Since 8.1 would have been out leaving PHP and browser support to mature a little).

HEIC isn't really viable, especially for our use where the image system is primarily for images to be visible within content and the gallery. Think it's going to be a while (Range of years potentially) before we see viability here.

**Comment by Nawor3565** [comment]
@ssddanbrown Any updates on AVIF support? When trying to upload an AVIF image, it's actually displayed in the pop-up message saying that the format isn't supported, implying that AVIFs are already functional but not enabled.

![Image](https://github.com/user-attachments/assets/d57a6606-0fb3-47ca-9b6f-3a76bc0d2c3c)

**Comment by ssddanbrown** [comment]
@Nawor3565 No updates but it looks well supported now. Feel free to open a specific request for it, referencing this issue for context, so it's available for something to be picked up as a potential easy add for a future feature release.

**Comment by prohtex** [comment]
+1 for HEIC support as it is so popular with iOS users.

> ### HEIC
> 
> * System:
>   
>   * [Not yet supported by GD](https://github.com/Intervention/image/blob/a1abc986fd644365e1740e70ea89b43b39c5cd59/src/Intervention/Image/Gd/Encoder.php#L174-L179)

@ssddanbrown have you considered VIPS? Performance and overall support for formats far exceeds GD and Imagemagick in my experience. I really love working with that library.

**Comment by ssddanbrown** [comment]
@prohtex I'd generally not be keen on adding extra requirements/dependencies for something of minor demand, and [browser support](https://caniuse.com/?search=heic) is still bad enough to make our support questionable, although at least Apple now seem to support their own format.

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/2505
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
